model-server/src/main/kotlin/org/modelix/model/server/handlers/IRepositoriesManager.kt

```diff
@@ -17,46 +17,77 @@
 
+/**
+ * Exposes high-level methods to interact with repositories, their branches and their versions.
+ *
+ * All methods are supposed to use non-blocking IO (or dispatch it appropriately).
+ */
 interface IRepositoriesManager {
+    suspend fun getRepositories(): Set<RepositoryId>
+
     /**
-     * Used to retrieve the server ID. If needed, the server ID is created and stored.
-     *
-     * If a server ID was not created yet, it is generated and saved in the database.
-     * It gets stored under the current and all legacy database keys.
-     *
-     * If the server ID was created previously but is only stored under a legacy database key,
-     * it also gets stored under the current and all legacy database keys.
+     * Creates a repository or throws an [RepositoryAlreadyExistsException].
+     * Returns the created initial version of the repository.
+     * The initial version has [userName] as [CLVersion.author]
+     * [useRoleIds] is used when creating the [CLTree] for the initial version.
      */
-    suspend fun maybeInitAndGetSeverId(): String
-    fun getRepositories(): Set<RepositoryId>
     suspend fun createRepository(repositoryId: RepositoryId, userName: String?, useRoleIds: Boolean = true): CLVersion
-    suspend fun removeRepository(repository: RepositoryId): Boolean
-
-    fun getBranches(repositoryId: RepositoryId): Set<BranchReference>
 
+    /**
+     * Removes the repository.
+     * That data is not yet removed from the database.
+     * See https://issues.modelix.org/issue/MODELIX-856
+     * Returns true if the repository existed and was removed.
+     * Returns false if the repository did not exist.
+     */
+    suspend fun removeRepository(repository: RepositoryId): Boolean
+    suspend fun getBranches(repositoryId: RepositoryId): Set<BranchReference>
     suspend fun removeBranches(repository: RepositoryId, branchNames: Set<String>)
 
     /**
-     * Same as [removeBranches] but blocking.
-     * Caller is expected to execute it outside the request thread.
+     * Returns the current version in this branch or null if the branch does not exist.
      */
-    fun removeBranchesBlocking(repository: RepositoryId, branchNames: Set<String>)
     suspend fun getVersion(branch: BranchReference): CLVersion?
+
+    /**
+     * Returns the hash of the current version in this branch or null if the branch does not exist.
+     */
     suspend fun getVersionHash(branch: BranchReference): String?
+
+    /**
+     * Returns the hash of the current version in this [branch] or null if the branch does not exist.
+     *
+     * The function executes immediately if the latest known version [lastKnown] is specified
+     * and the current version differs.
+     * Else the function delays the coroutine waiting for a new version on the [branch] and returns its hash.
+     * The delay is limited with a timeout after which the hash of the current version is returned
+     * even if no new version was created in the meantime.
```

Comment:
We probably later have to think about this method again. Having a method that uses an unknown timeout is a bit weird as the interface.

Reply:
It is tricky. But `RepositoriesManager` is still just an internal and model server specific class, so this should be fine for now.